### ISSUE TITLE
Lisp process never returns when running "(parse-js:parse-js nil)"

### ISSUE THREAD

**Issue by rolpereira** [marijnh/parse-js#7]
Hello,

If I execute the form `(parse-js:parse-js nil)` the REPL never returns a value.

This is the backtrace given by sbcl if I press C-c:

```
Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {100298C9D3}>
0: ("bogus stack frame")
1: (SB-SYS:WAIT-UNTIL-FD-USABLE 0 :INPUT NIL T)
2: (SB-IMPL::REFILL-INPUT-BUFFER #<SB-SYS:FD-STREAM for "standard input" {1002D94C13}>)
3: (SB-IMPL::INPUT-CHAR/UTF-8 #<SB-SYS:FD-STREAM for "standard input" {1002D94C13}> NIL :EOF)
4: ((LAMBDA (&REST REST) :IN SB-IMPL::GET-EXTERNAL-FORMAT) #<SB-SYS:FD-STREAM for "standard input" {1002D94C13}> NIL :EOF)
5: (READ-CHAR #<SB-SYS:FD-STREAM for "standard input" {1002D94C13}> NIL :EOF #<unused argument>)
6: (PEEK-CHAR NIL NIL NIL NIL #<unused argument>)
7: ((LABELS PARSE-JS::NEXT-TOKEN :IN PARSE-JS:LEX-JS) NIL)
8: (PARSE-JS::PARSE-JS* NIL NIL)
9: (PARSE-JS:PARSE-JS NIL :STRICT-SEMICOLONS NIL :ECMA-VERSION 3 :RESERVED-WORDS NIL)
10: (SB-INT:SIMPLE-EVAL-IN-LEXENV (PARSE-JS:PARSE-JS NIL) #<NULL-LEXENV>)
11: (EVAL (PARSE-JS:PARSE-JS NIL))
12: (INTERACTIVE-EVAL (PARSE-JS:PARSE-JS NIL) :EVAL NIL)
13: (SB-IMPL::REPL-FUN NIL)
14: ((LAMBDA NIL :IN SB-IMPL::TOPLEVEL-REPL))
15: (SB-IMPL::%WITH-REBOUND-IO-SYNTAX #<CLOSURE (LAMBDA NIL :IN SB-IMPL::TOPLEVEL-REPL) {1006A3C34B}>)
16: (SB-IMPL::TOPLEVEL-REPL NIL)
17: (SB-IMPL::TOPLEVEL-INIT)
18: ((FLET #:WITHOUT-INTERRUPTS-BODY-84 :IN SAVE-LISP-AND-DIE))
19: ((LABELS SB-IMPL::RESTART-LISP :IN SAVE-LISP-AND-DIE))
```

And this is the transcript that shows how to trigger the issue:

```
This is SBCL 1.3.3, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
* (ql:quickload :parse-js)
To load "parse-js":
  Load 1 ASDF system:
    parse-js
; Loading "parse-js"

(:PARSE-JS)
* (parse-js:parse-js nil)
^C
debugger invoked on a SB-SYS:INTERACTIVE-INTERRUPT in thread
#<THREAD "main thread" RUNNING {100298C9D3}>:
  Interactive interrupt at #x7FFFF73E08C0.

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [CONTINUE] Return from SB-UNIX:SIGINT.
  1: [ABORT   ] Exit debugger, returning to top level.

("bogus stack frame")
0] backtrace

```
A way to remove this issue would be to throw a `js-parse-error` condition.
Either that or return `(:TOPLEVEL NIL)` as done by `(parse-js:parse-js "")`

Regards

**Comment by marijnh** [comment]
`parse-js` expects a string or a stream. If you're going to pass it a symbol, all bets are off. I'm not really interested in handling bogus arguments specially, but if you want to submit a PR that adds an `etypecase` or `check-type` to ensure an error, I'd probably merge that.

**Comment by rolpereira** [comment]
Hello,

I create a pull request that fixes this issue in `parse-js:parse-js`.

I also found a similar issue in `parse-js:read-js-number` and added a similar check to it.

**Comment by marijnh** [comment]
Fixed via #8
